### ISSUE TITLE
Feature request: add possibility to specify a values file inside a packaged chart

### ISSUE THREAD

**Issue by yann-soubeyrand** [helm/helm#4782]
Hi!

Oftentimes, one need to have values specific to the environment where the chart will be deployed. A possible solution is to have a common values.yaml file and several values-\<environment\>.yaml files. Deploying a chart from the source directory is easy using helm install --values values-\<environment\>.yaml.

However, when installing a packaged chart (from a chart repository), one cannot specify to use the values-\<environment\>.yaml file packaged inside the chart. This feature request is about adding the possibility to do so.

A better solution would be to add the notion of environment as a first class citizen in Helm and automatically use the values-\<environment\>.yaml file if present, but this would involve a lot more work!

**Comment by hickeyma** [comment]
How about `helm fetch --untar` the chart before passing the values file?

**Comment by yann-soubeyrand** [comment]
> How about `helm fetch --untar` the chart before passing the values file?

This is a possible solution but not very user friendly ;-)

**Comment by maorfr** [comment]
Hey @hickeyma ,

We wrote a tool that wraps helm and has a `deploy` command that does all the logic for you.
Check out the docs:
https://github.com/maorfr/orca/tree/master/docs/commands#deploy-chart

Could this be of any help for you?

**Comment by hickeyma** [comment]
@maorfr Thanks for this.

@yann-soubeyrand could this tool be of use?

**Comment by yann-soubeyrand** [comment]
Yes, this tool may bring a solution to our use case, but having to use another tool "just" for this is not ideal in my opinion ;-) I think this should be a feature of Helm itself. Anyway, I'll have an eye on Orca to see if we can benefit from other features of the tool as well.

**Comment by hickeyma** [comment]
Sure. I am not a core maintainer but @bacongobbler who is, has added the "feature" tag to this, so it might make it into a release at some stage.

**Comment by tamalsaha** [comment]
cc: @mattfarina 

I propose that the following syntax to load values file from inside a chart archive.

```
--values chart:///values-production.yaml
```

This was all the existing behavior can be preserved. Do you think this is acceptable to add in Helm 3? If yes, I can try to send a pr.

**Comment by bacongobbler** [comment]
the URL syntax looks wonky, but the premise sounds fine. I'd also suggest looking at other tickets asking for the same feature. Namely #6715

**Comment by fritzb** [comment]
@bacongobbler any update on this feature ? Referring to a specific value file inside a chart should be part of basic functionality IMO. It is going to be closed soon and marked as stale (similar to https://github.com/helm/helm/issues/6715)

**Comment by bobbui** [comment]
bump!

**Comment by MaxRuenz** [comment]
bump!

**Comment by bobbui** [comment]
bump!

**Comment by soujiro32167** [comment]
yes please!

**Comment by dhanvi** [comment]
bump

**Comment by bacongobbler** [comment]
Since others have been asking... No progress has been made on this PR by members of the community.

The first step to propose a new feature would be to write a HIP.

We intend HIPs to be the primary mechanisms for proposing major new features, for collecting community input on an issue, and for documenting the design decisions that have gone into the project. The proposal author is responsible for building consensus within the community and documenting dissenting opinions.

See [Proposing an Idea](https://github.com/helm/helm/blob/main/CONTRIBUTING.md#proposing-an-idea) for more information on this process.

**Comment by soujiro32167** [comment]
bump

**Comment by dhanvi** [comment]
bump :)

**Comment by dhanvi** [comment]
bump

**Comment by ps-hdavoodi** [comment]
bump!

**Comment by saireddyb** [comment]
Is this feature planned to be implemented ?

**Comment by ncaro-fdz** [comment]
bump

**Comment by zamonia500** [comment]
bump

**Comment by tw-sematell** [comment]
bump

**Comment by dhanvi** [comment]
bump

**Comment by thiagowfx** [comment]
Could a core maintainer please tag this with the "keep open" label so that it isn't closed accidentally in the future, and also so that there is no need for folks to keep bumping it so that it stays open? ([c.f.](https://github.com/helm/helm/blob/f5e7b117c552e553bc282e7c3b9d111f410324d7/.github/workflows/stale-issue-bot.yaml#L16))  Thanks!

**Comment by dhanvi** [comment]
bump

**Comment by terry-brown-26870** [comment]
bump
